### DISCUSSION TITLE
Does imodeljs support renderschedule and timeline ?

### DISCUSSION THREAD

**Discussion by namsancow** [iTwin/itwinjs-core#1384] [Q&A] [answered]
I saw the interesting iTwin Developer Webinar posted by Bentley iTwin Platform on youtube.
https://youtu.be/zFL8pCm1UXA

In this video, the renderschedule and timeline is explained for rotating wind turbine.
Does imodeljs also support renderschedule and timeline as like this video ?

There are two open source versions as follows, which is confusing. 
1) github.com/imodeljs
2) github.com/iTwin

I am also curious if imodeljs will be changed to iTwin.

**Comment by pmconne** [comment, marked as answer]
The app in that video is an iModel.js-based app, so yes - the display system in [imodeljs-frontend](https://github.com/imodeljs/imodeljs/tree/master/core/frontend) animates the view according to its schedule script.

You can use a [ScriptBuilder](https://github.com/imodeljs/imodeljs/blob/3de4eafd3ab6e181b54d771b22cfed762469c62b/core/common/src/RenderSchedule.ts#L1086) to produce a [Script](https://github.com/imodeljs/imodeljs/blob/3de4eafd3ab6e181b54d771b22cfed762469c62b/core/common/src/RenderSchedule.ts#L798) that can be persisted on a [RenderTimeline](https://github.com/imodeljs/imodeljs/blob/3de4eafd3ab6e181b54d771b22cfed762469c62b/core/backend/src/Element.ts#L1569) element to which a [DisplayStyle](https://github.com/imodeljs/imodeljs/blob/3de4eafd3ab6e181b54d771b22cfed762469c62b/core/common/src/DisplayStyleSettings.ts#L626) can refer. Animate the view by manipulating the [Viewport's time point](https://github.com/imodeljs/imodeljs/blob/3de4eafd3ab6e181b54d771b22cfed762469c62b/core/frontend/src/Viewport.ts#L466), perhaps [using this](https://github.com/imodeljs/imodeljs/blob/master/ui/framework/src/ui-framework/timeline/ScheduleAnimationProvider.ts).

There is only one [iModel.js](https://github.com/imodeljs/imodeljs) library. Most of the other repositories in those two GitHub projects are examples of apps that use iModel.js, or packages that consume iModel.js. I don't know the details of why there are two projects. We don't intend to change the name of the imodeljs repository, but iModels are only one part of a digital twin so "iTwin" is the more general term.

**Reply by namsancow** [reply]
Many thanks to your advice.
I've always been looking forward to simple animation features that can bring the model space to life, 
and I'm very excited to know the ScriptBuilder and the RenderTimeline.
Best regards
